src/cpu/flexcpu/simple_dataflow_thread.hh

```diff
@@ -0,0 +1,248 @@
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are
+ * met: redistributions of source code must retain the above copyright
+ * notice, this list of conditions and the following disclaimer;
+ * redistributions in binary form must reproduce the above copyright
+ * notice, this list of conditions and the following disclaimer in the
+ * documentation and/or other materials provided with the distribution;
+ * neither the name of the copyright holders nor the names of its
+ * contributors may be used to endorse or promote products derived from
+ * this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ * Authors: Bradley Wang
+ */
+
+#ifndef __CPU_FLEXCPU_SIMPLE_DATAFLOW_THREAD_HH__
+#define __CPU_FLEXCPU_SIMPLE_DATAFLOW_THREAD_HH__
+
+#include <memory>
+#include <set>
+#include <unordered_map>
+#include <vector>
+
+#include "cpu/exec_context.hh"
+#include "cpu/flexcpu/generic_reg.hh"
+#include "cpu/flexcpu/inflight_inst.hh"
+#include "cpu/flexcpu/simple_dataflow_cpu.hh"
+#include "cpu/inst_seq.hh"
+#include "cpu/reg_class.hh"
+#include "cpu/simple_thread.hh"
+#include "mem/request.hh"
+
+
+/**
+ * Instances of this class represent a logical "Core" within the
+ * SimpleDataflowCPU.
+ */
+class SDCPUThread
+{
+  protected:
+    // BEGIN SDCPUThread Internal variables
+
+    /**
+     * We hold a pointer to the CPU which owns this logical thread.
+     */
+    SimpleDataflowCPU* _cpuPtr;
+
+    /**
+     * The SimpleThread class already contains all internal state necessary to
+     * hold onto the state of a basic CPU, demonstrated by its use in many
+     * other CPU models. Until we find a significant need to hold particular
+     * permanent committed state different from a regular CPU in this library,
+     * there is no reason to redefine the wheel.
+     *
+     * The committed state represents the state resulting from the complete
+     * correct execution of all instructions up to the point represented by
+     * numInstsCommitted.
+     */
+    SimpleThread* _committedState;
```

Comment:
Not a pointer?

Reply:
So I tried that, but SimpleThread didn't seem to like it when I first did it... I could mess with it again but that's what had been working so far.